packages/node_modules/@ciscospark/spark-core/test/unit/spec/interceptors/rate-limit.js

```diff
@@ -0,0 +1,291 @@
+          },
+          config: cloneDeep(config)
+        });
+
+        spark.credentials.supertoken = new Token({
+          access_token: `ST1`,
+          token_type: `Bearer`
+        }, {parent: spark});
+
+        interceptor = Reflect.apply(RateLimitInterceptor.create, spark, []);
+      });
+
+      describe(`#isRateLimited`, () => {
+        it(`returns false when URI is not IDbroker URI`, () => {
+          assert.equal(interceptor.isRateLimited(`https://fakeuir.com/testFake`), false);
+        });
+        it(`returns false if the URI is null`, () => {
+          assert.equal(interceptor.isRateLimited(null), false);
+        });
+        it(`returns true when idbroker API is rate limited`, () => {
+          interceptor.setRateLimitExpiry(`https://idbroker.webex.com/horse/v1/myID`, new Date().getTime() * 2);
+          assert.equal(interceptor.isRateLimited(`https://idbroker.webex.com/horse/v1/myID`), true);
+        });
+        it(`returns true when identity API is rate limited`, () => {
+          interceptor.setRateLimitExpiry(`https://identity.webex.com/horse/v1/myID`, new Date().getTime() * 2);
+          assert.equal(interceptor.isRateLimited(`https://identity.webex.com/horse/v1/myID`), true);
+        });
+        it(`returns false when previously rate limited idbroker API expiry time has been met`, () => {
+          interceptor.setRateLimitExpiry(`https://idbroker.webex.com/horse/v1/myID`, 1);
+          return sleep(2000)
+           .then(() => {
+             assert.equal(interceptor.isRateLimited(`https://idbroker.webex.com/horse/v1/myID`), false);
+           });
+        });
+        it(`returns false when previously rate limited identity API expiry time has been met`, () => {
+          interceptor.setRateLimitExpiry(`https://identity.webex.com/horse/v1/myID`, 1);
+          return sleep(2000)
+           .then(() => {
+             assert.equal(interceptor.isRateLimited(`https://identity.webex.com/horse/v1/myID`), false);
+           });
+        });
+      });
+
+      describe(`#getApiName`, () => {
+        it(`returns null when there is no regex match`, () => {
+          assert.equal(interceptor.getApiName(`https://fakeuir.com/testFake`), null);
+        });
+        it(`returns null when the argument is null`, () => {
+          assert.equal(interceptor.getApiName(null), null);
+        });
+        it(`returns idbroker API name if there is a match`, () => {
+          assert.equal(interceptor.getApiName(`https://idbroker.webex.com/horse/v1/myID`), `horse`);
+        });
+        it(`returns identity API name if there is a match`, () => {
+          assert.equal(interceptor.getApiName(`https://identity.webex.com/horse/v1/myID`), `horse`);
+        });
+      });
+
+      describe(`#getRateLimitStatus`, () => {
+        it(`returns false if API name is not in rate limit expiry map`, () => {
+          assert.equal(interceptor.getRateLimitStatus(`https://fakeuir.com/testFake`), false);
```

Comment:
Let's use `example.com` for all domains we don't control. Yes, these are unit tests, but we're using a lot of real domains for testing throughout the product that have definitely caused wierd issues.